data-prepper-plugins/saas-source-plugins/saas-crawler/src/main/java/org/opensearch/dataprepper/plugins/source/saas_crawler/base/SaasSourcePlugin.java

```diff
@@ -0,0 +1,113 @@
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSetManager;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.codec.ByteDecoder;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.plugin.PluginFactory;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.Source;
+import org.opensearch.dataprepper.model.source.coordinator.SourcePartitionStoreItem;
+import org.opensearch.dataprepper.model.source.coordinator.enhanced.EnhancedSourceCoordinator;
+import org.opensearch.dataprepper.model.source.coordinator.enhanced.EnhancedSourcePartition;
+import org.opensearch.dataprepper.model.source.coordinator.enhanced.UsesEnhancedSourceCoordination;
+import org.opensearch.dataprepper.plugins.source.saas_crawler.coordination.PartitionFactory;
+import org.opensearch.dataprepper.plugins.source.saas_crawler.coordination.partition.LeaderPartition;
+import org.opensearch.dataprepper.plugins.source.saas_crawler.coordination.scheduler.LeaderScheduler;
+import org.opensearch.dataprepper.plugins.source.saas_crawler.coordination.scheduler.WorkerScheduler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Function;
+
+
+
+/**
+ * JiraConnector connector entry point.
+ */
+
+public abstract class SaasSourcePlugin implements Source<Record<Event>>, UsesEnhancedSourceCoordination {
+
+
+  private static final Logger log = LoggerFactory.getLogger(SaasSourcePlugin.class);
+  private final PluginMetrics pluginMetrics;
+  private final PluginFactory pluginFactory;
+
+  private final AcknowledgementSetManager acknowledgementSetManager;
+
+  private final ExecutorService executorService;
+
+  private EnhancedSourceCoordinator coordinator;
+
+  @Getter
+  private final SaasSourceConfig sourceConfig;
+
+  private Buffer<Record<Event>> buffer;
+  private final Crawler crawler;
+
+
+  public SaasSourcePlugin(final PluginMetrics pluginMetrics,
+                          final SaasSourceConfig sourceConfig,
+                          final PluginFactory pluginFactory,
+                          final AcknowledgementSetManager acknowledgementSetManager,
+                          final Crawler crawler,
+                          final SaasPluginExecutorServiceProvider executorServiceProvider) {
+    log.info("Create SaaS Source Connector");
+    this.pluginMetrics = pluginMetrics;
+    this.sourceConfig = sourceConfig;
+    this.pluginFactory = pluginFactory;
+    this.crawler = crawler;
+
+    this.acknowledgementSetManager = acknowledgementSetManager;
+    this.executorService = executorServiceProvider.get();
+  }
+
+
+  @Override
+  public void start(Buffer<Record<Event>> buffer) {
+    Objects.requireNonNull(coordinator);
+    log.info("Starting SaaS Source Plugin... ");
```

Comment:
Let's avoid ellipses in logs.

```suggestion
    log.info("Starting SaaS Source Plugin.");
```

Also, can we get some additional information about the source here?

Reply:
What is a user of Data Prepper to make of "SaaS Source Plugin"? It is too generic. We need logs that say something more like "Starting Jira source plugin."

Maybe the name should be a parameter in the constructor.

Reply:
Added a name parameter and modified the log statement to print the name